### PR TITLE
#3: Guidelines for contributing to open source

### LINKED ISSUE #3: Create guidelines for developers who want to contribute to open source
Some developers are experienced with OSS, some are not. We want to create a guide to answer the questions a Visma developer may have when they consider contributing to open source projects, e.g.

1) What can/should I contribute to?
2) Do I need to use any particular name or account?
3) How do I ensure I am a good OSS citizen?

### PR DESCRIPTION
**Pull request by petli**

### CODE REVIEW

**Comment by reshadf** [comment]
I think this is a good first setup for this document. We could merge it and patch/rewrite it when necessary if we come up with new guidelines. What do you think?

File: contributing-to-open-source.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+its future direction, you may want to start to monitor the pull
+requests that come in and review the ones that touch on your
+interests.  QAs can help test PRs too.
+
+
+# Maintaining open source projects
+
+If you start being visible in projects by submitting code, discussing
+issues and reviewing PRs, you may be asked by to become a maintainer.
+This typically comes with push rights to the main repository and clout
```
Comment (idrydi) [inline contributing-to-open-source.md]: What is "clout"?

File: contributing-to-open-source.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+in shaping the projects roadmap, but also expectations that you will
+review PRs and help with fixing bugs.
+
+It is a committment, and you need to discuss in your team and your
```
Comment (idrydi) [inline contributing-to-open-source.md]: Commitment, with one t

File: contributing-to-open-source.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+Once you have some significant code in a project or really care about
+its future direction, you may want to start to monitor the pull
+requests that come in and review the ones that touch on your
+interests.  QAs can help test PRs too.
+
+
+# Maintaining open source projects
+
+If you start being visible in projects by submitting code, discussing
+issues and reviewing PRs, you may be asked by to become a maintainer.
```
Comment (idrydi) [inline contributing-to-open-source.md]: Remove "by"

File: contributing-to-open-source.md [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+review PRs and help with fixing bugs.
+
+It is a committment, and you need to discuss in your team and your
+manager how much time you may have to devote to this, and if you have
+that time.
+
+Think about if you have a special focus area, and make it clear to the
+other maintainers if you will prioritise that and not look much on
+other parts of the project.  Also be clear about if you will leave the
+mainatiner role if you leave the Visma project, or stay on.
```
Comment (idrydi) [inline contributing-to-open-source.md]: Maintainer

File: contributing-to-open-source.md [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+our every day work: git and Github, branching, pushing code,
+submitting pull requests, reviewing, testing and merging.
+
+The rest of this document discusses the different ways you can
+contribute to open source projects as a Visma employee.
+
+The advice is also useful for extra-curricular contributions to open
+source that is not directly related to what you do at Visma.  This is
+a great way to learn new programming languages and frameworks.  To
+help you get started many projects have issues labled "up-for-grabs"
```
Comment (idrydi) [inline contributing-to-open-source.md]: Labeled

File: contributing-to-open-source.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+possible test it, and provide feedback.
+
+
+# Bug fixes
+
+Think you can fix a bug you've found in an open source project?  Start
+by reporting it as described above, and add a note that you will try
+to fix it yourself.
+
+Fixing bugs in open source is not much different to doing in your
```
Comment (idrydi) [inline contributing-to-open-source.md]: Maybe "...not much different than in your normal projects, but...".
Not sure about the use of normal, but couldn't think of anything better myself.

File: contributing-to-open-source.md [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+  information that is important to maintain the quality of the code.
+  Read it before you start.  Sometimes this is in the `README`
+  instead.
+
+- If that document doesn't say it, try to figure out which code branch
+  to work against.  Do they want fixes against `main`, `dev` or a
+  release branch?
+
+- Reference the issue ID number as `#123` in the commit messages to
+  link them and the pull request to he issue.
```
Comment (idrydi) [inline contributing-to-open-source.md]: Missing a t in "the"

**Review by idrydi — CHANGES_REQUESTED** [review]
Found a few minor misspellings

File: contributing-to-open-source.md (hunk 1 above)
Comment (petli) [reply contributing-to-open-source.md]: https://www.merriam-webster.com/dictionary/clout meaning 4 :)

Comment (petli) [reply contributing-to-open-source.md]: "influence over" is a better phrase, in any case.

**Review by idrydi — APPROVED** [review]

**Pull request merged**